cairo/tests/utils/serde.py

```diff
@@ -155,7 +155,7 @@ def __init__(
         cairo_file=None,
     ):
         self.segments = segments
-        self.memory = segments.memory
+        self.memory = segments.memory if segments else None
```

Comment:
i know it's kind of an anti pattern and code smell to do this but frankly I don't really want to dedicate more time to this issue